### PR TITLE
NO-ISSUE: shift week - Add event subscription API

### PR DESCRIPTION
**Pull request by eranco74**
Shift week - don't review

### CODE REVIEW

File: swagger.yaml [hunk 1]
```diff
@@ -3393,6 +3393,165 @@ paths:
           schema:
             $ref: '#/definitions/error'
 
+  /v2/events/subscription:
+    post:
+      tags:
+        - events
+      security:
+        - userAuth: [admin, read-only-admin, user]
+        - urlAuth: []
```
Comment (avishayt) [inline swagger.yaml:3402]:
```suggestion
```

File: swagger.yaml [hunk 2]
```diff
@@ -3393,6 +3393,165 @@ paths:
...
+      description: Subscribe a URL to be called upon events for a cluster.
```
Comment (jkilzi) [inline swagger.yaml:3403]:
```suggestion
      description: Subscribe a URL to be called when events for a cluster are emitted.
```
:)

Comment (avishayt) [inline swagger.yaml:3403]:
```suggestion
      description: Subscribe to an event type with URL callback.
```

File: swagger.yaml [hunk 3]
```diff
@@ -3393,6 +3393,165 @@ paths:
...
+      operationId: v2EventsSubscribe
+      parameters:
+        - in: body
+          name: new-cluster-params
```
Comment (jkilzi) [inline swagger.yaml:3407]: ?

Comment (avishayt) [inline swagger.yaml:3407]:
```suggestion
          name: new-event-subscription-params
```

Comment (jkilzi) [reply swagger.yaml:3407]: shouldn't it be `event-subscription-create`?

File: swagger.yaml [hunk 4]
```diff
@@ -3393,6 +3393,165 @@ paths:
...
+          description: Error.
+          schema:
+            $ref: '#/definitions/error'
+    get:
+      tags:
+        - events
+      security:
+        - userAuth: [admin, read-only-admin, user]
+        - urlAuth: []
+      description: Get events subscriptions.
```
Comment (avishayt) [inline swagger.yaml:3443]:
```suggestion
      description: Get event subscriptions.
```

File: swagger.yaml [hunk 5]
```diff
@@ -3393,6 +3393,165 @@ paths:
...
+      operationId: v2EventsSubscriptionList
+      parameters:
+        - in: query
+          name: cluster_id
+          description: The cluster id.
```
Comment (avishayt) [inline swagger.yaml:3448]:
```suggestion
          description: The cluster to return subscriptions for.
```

File: swagger.yaml [hunk 6]
```diff
@@ -3393,6 +3393,165 @@ paths:
...
+          schema:
+            $ref: '#/definitions/error'
+
+    delete:
+      tags:
+        - events
+      security:
+        - userAuth: [ admin, read-only-admin, user ]
+        - urlAuth: [ ]
+      description: Delete events subscription.
```
Comment (avishayt) [inline swagger.yaml:3522]:
```suggestion
      description: Delete event subscription.
```

File: swagger.yaml [hunk 7]
```diff
@@ -3393,6 +3393,165 @@ paths:
...
+            $ref: '#/definitions/error'
+
+  /v2/events/subscription/{subscription_id}:
+    get:
+      tags:
+        - events
+      security:
+        - userAuth: [admin, read-only-admin, user]
+        - urlAuth: []
+      description: Get events subscription.
```
Comment (avishayt) [inline swagger.yaml:3481]:
```suggestion
      description: Get an event subscription.
```

File: swagger.yaml [hunk 8]
```diff
@@ -3393,6 +3393,165 @@ paths:
...
+          schema:
+            $ref: '#/definitions/error'
+
+  /v2/events/subscription/{subscription_id}:
+    get:
+      tags:
+        - events
+      security:
+        - userAuth: [admin, read-only-admin, user]
+        - urlAuth: []
```
Comment (avishayt) [inline swagger.yaml:3480]:
```suggestion
```

File: swagger.yaml [hunk 9]
```diff
@@ -3393,6 +3393,165 @@ paths:
...
+          description: Error.
+          schema:
+            $ref: '#/definitions/error'
+
+    delete:
+      tags:
+        - events
+      security:
+        - userAuth: [ admin, read-only-admin, user ]
+        - urlAuth: [ ]
```
Comment (avishayt) [inline swagger.yaml:3521]:
```suggestion
```

File: swagger.yaml [hunk 10]
```diff
@@ -3393,6 +3393,165 @@ paths:
...
+        "500":
+          description: Error.
+          schema:
+            $ref: '#/definitions/error'
+    get:
+      tags:
+        - events
+      security:
+        - userAuth: [admin, read-only-admin, user]
+        - urlAuth: []
```
Comment (avishayt) [inline swagger.yaml:3442]:
```suggestion
```

**Pull request closed without merging**